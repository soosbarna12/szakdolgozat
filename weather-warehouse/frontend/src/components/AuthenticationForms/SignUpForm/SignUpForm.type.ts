export interface SignUpFormProps {
  open: boolean;
  onClose: () => void;
  onRegisterSuccess?: () => void;
}
