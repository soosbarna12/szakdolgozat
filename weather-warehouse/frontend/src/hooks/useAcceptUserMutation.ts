import { useMutation } from "@tanstack/react-query";
import { useAlert } from "../utils/AlertContext";
import axios from "../utils/axiosConfig";


export function useAcceptUserMutation() {
  const { showAlert } = useAlert();

  const mutation = useMutation({
    mutationFn: async (id: number) => {
      const response = await axios.post(`/user/accept`, { id });
      return response.data;
    },
    onSuccess: () => {
      showAlert("User accepted successfully", "success");
    },
    onError: () => {
      showAlert("Failed to accept user", "error");
    },
  });

  return mutation;
}
