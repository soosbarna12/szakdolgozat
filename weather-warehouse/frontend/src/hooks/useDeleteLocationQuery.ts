import { useMutation } from "@tanstack/react-query";
import { useAlert } from "../utils/AlertContext";
import axios from "../utils/axiosConfig";

export function useDeleteLocationQuery() {
  const { showAlert } = useAlert();

  const mutation = useMutation({
    mutationFn: async (userLocationID: number) => {
      const response = await axios.delete(`/user/deleteLocation`, { params: { userLocationID } });
      return response.data;
    },
    onSuccess: () => {
      showAlert("Location deleted successfully", "success");
    },
    onError: () => {
      showAlert("Failed to delete location", "error");
    },
  });

  return mutation;
}