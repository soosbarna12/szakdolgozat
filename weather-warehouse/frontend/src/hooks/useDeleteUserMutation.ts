import { useMutation } from "@tanstack/react-query";
import { useAlert } from "../utils/AlertContext";
import axios from "../utils/axiosConfig";


export function useDeleteUserMutation() {
  const { showAlert } = useAlert();

  const mutation = useMutation({
    mutationFn: async (id: number) => {
      const response = await axios.delete(`/user/delete`, { params: { id } });
      return response.data;
    },
    onSuccess: () => {
      showAlert('User deleted successfully', 'success');
    },
    onError: () => {
      showAlert('Failed to delete user', 'error');
    },
  });

  return mutation;
}
