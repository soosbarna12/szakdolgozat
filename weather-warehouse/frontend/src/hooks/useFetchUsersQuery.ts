import { useQuery } from "@tanstack/react-query";
import { User } from "../types/userData.type";
import axios from "../utils/axiosConfig";


export function useFetchUsersQuery() {
  
  const { data, error, isLoading, refetch } = useQuery<User[]>({
    queryKey: ["users"],
    queryFn: async () => {
      const response = await axios.get(`/api//user/userData`);

      return response.data as User[];
    },
  });

  return { data, error, isLoading, refetch };
}
