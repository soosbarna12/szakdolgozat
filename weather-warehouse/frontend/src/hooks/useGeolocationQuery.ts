import { useQuery } from "@tanstack/react-query";
import axios from "../utils/axiosConfig";
import { Pages } from "../types/page.type";


/*
export function useGeolocationQuery(location: string) {
    
  const { data, error, isLoading } = useQuery({
    queryKey: ["location", location],
    queryFn: async () => {
      const response = await axios.get(`/geo/location`, {
        params: { location }
      })

      if (!response?.data || response?.data?.length === 0) {
        return []; // return empty array if no data found
      }

      // search for unique locations
      const newResults = response.data.filter(
        (loc: any, index: number, self: any[]) =>
          index ===
          self.findIndex(
            (t) => t.name === loc.name && t.state === loc.state && t.country === loc.country
          )
      );
      return newResults; 
    },

    enabled: !!location,
  });

  return { data, error, isLoading };
}
*/

export function useGeolocationQuery(locationName: string, type: Pages) {
  const { data, error, isLoading } = useQuery({
    queryKey: ["locationName", locationName],
    queryFn: async () => {
      if (!locationName.trim()) {
        return []; // Return an empty array if no city name is provided
      }

      try {
        const response = await axios.get(`/geo/location`, {
          params: {
            location: locationName.trim(),
           },
        });

        if (!response?.data || response?.data?.length === 0) {
          return []; // Return an empty array if no data is found
        }

        // Filter unique city results
        const uniqueResults = response.data.filter(
          (loc: any, index: number, self: any[]) =>
            index ===
            self.findIndex(
              (t) => t.name === loc.name && t.state === loc.state && t.country === loc.country
            )
        );
        //console.log("Unique results:", uniqueResults); // Log the unique results for debugging

        return uniqueResults;
      } catch (error) {
        console.error("Error fetching geolocation data:", error);
        throw new Error("Failed to fetch geolocation data");
      }
    },
    enabled: (locationName.length !== 0 && type === Pages.Today),
    refetchOnWindowFocus: false, // Prevent refetching when the window regains focus
  });

  return { data, error, isLoading };
}