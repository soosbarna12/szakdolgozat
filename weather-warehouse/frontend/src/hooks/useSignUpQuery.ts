import { useQuery } from "@tanstack/react-query";
import { useAlert } from "../utils/AlertContext";
import axios from "../utils/axiosConfig";
import { useEffect } from "react";


export function useSignUpQuery(username: string, password: string, securityQuestion: string, securityAnswer: string) {
  const { showAlert } = useAlert();
  
  const { data: registerData, error, isLoading, isSuccess, refetch } = useQuery({
    queryKey: ["register", username],
    queryFn: async () => {
      const response = await axios.post(`/user/register`, {
          username,
          password,
          securityQuestion,
          securityAnswer,
        });

      return response.data;
    },
      refetchOnWindowFocus: false,
      retry: false,
      enabled: false, // Disable automatic refetching, we will call refetch() manually -> to use this query when clicking the login button
  });

  useEffect(() => {
    if (isSuccess) {
      showAlert("User registered successfully", "success");
    }
  }, [isSuccess]);

  useEffect(() => {
    if (error) {
      showAlert("Registration failed", "error");
    }
  }, [error]);

  return { registerData, error, isLoading, isSuccess, refetch };
}
