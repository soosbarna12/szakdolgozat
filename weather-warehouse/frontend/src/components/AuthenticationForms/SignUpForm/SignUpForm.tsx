import { Visibility, VisibilityOff } from '@mui/icons-material';
import { DialogContent, IconButton, InputAdornment, Typography } from '@mui/material';
import * as React from 'react';
import { useEffect, useState } from 'react';
import { useSignUpQuery } from '../../../hooks/useSignUpQuery';
import { StyledButton } from '../../../stlyes/button.style';
import { StyledDialog } from '../../../stlyes/common.style';
import { StyledTextField } from '../../../stlyes/inputField.style';
import { SignUpFormProps } from './SignUpForm.type';


export function SignUpForm(props: Readonly<SignUpFormProps>) {
	const { open, onClose, onRegisterSuccess } = props;
	const [username, setUsername] = useState('');
	const [password, setPassword] = useState('');
	const [confirmPassword, setConfirmPassword] = useState('');
	const [securityQuestion, setSecurityQuestion] = useState('');
	const [securityAnswer, setSecurityAnswer] = useState('');
	const [showPassword, setShowPassword] = useState(false);
	const { isSuccess, refetch: refetchRegisterQuery } = useSignUpQuery(username, password, securityQuestion, securityAnswer);

	useEffect(() => {
		if (open) {
			setUsername('');
			setPassword('');
			setConfirmPassword('');
			setSecurityQuestion('');
			setSecurityAnswer('');
			setShowPassword(false);
		}
	}, [open]);

	// useffectbe kell rakni, mert ha a login sikeres, akkor a useLoginQuery hookban a isSuccess true lesz, viszont a refretch meghívása után nem fog egyből az isSuccess frissülni
	useEffect(() => {
		if (isSuccess) {
			onRegisterSuccess();
			//window.location.reload(); // refresh the page
			onClose();
		}
	}, [isSuccess])

	function handleClickShowPassword() {
		setShowPassword((show) => !show);
	}

	function handleMouseDownPassword(event: React.MouseEvent) {
		event.preventDefault();
	}

	function renderShowPassword() {
		return showPassword ? (
			<VisibilityOff fontSize="inherit" />
		) : (
			<Visibility fontSize="inherit" />
		);
	}

	function handleRegister() {
		refetchRegisterQuery(); // neccessary to use this query when clicking the login button
	};


	return (
		<StyledDialog open={open} onClose={onClose} maxWidth="xs" fullWidth data-testid="signUpDialog">
			<DialogContent sx={{ display: 'flex', flexDirection: 'column', gap: 1 }}>

				<Typography variant="h5" sx={{ textAlign: 'left' }} data-testid="signUpTitle">
					Sign Up
				</Typography>

				<StyledTextField
					name="username"
					type="text"
					placeholder="Username"
					required
					value={username}
					onChange={(e) => setUsername(e.target.value)}
					data-testid="usernameInput"
				/>

				<StyledTextField
					name="password"
					type={showPassword ? 'text' : 'password'}
					placeholder="Password"
					required
					value={password}
					onChange={(e) => setPassword(e.target.value)}
					endAdornment={
						<InputAdornment position="end">
							<IconButton
								aria-label="toggle password visibility"
								onClick={handleClickShowPassword}
								onMouseDown={handleMouseDownPassword}
								edge="end"
								data-testid="togglePasswordVisibility"
							>
								{renderShowPassword()}
							</IconButton>
						</InputAdornment>
					}
					data-testid="passwordInput"
				/>

				<StyledTextField
					name="confirmPassword"
					type={showPassword ? 'text' : 'password'}
					placeholder="Confirm Password"
					required
					value={confirmPassword}
					onChange={(e) => setConfirmPassword(e.target.value)}
					endAdornment={
						<InputAdornment position="end">
							<IconButton
								aria-label="toggle confirm password visibility"
								onClick={handleClickShowPassword}
								onMouseDown={handleMouseDownPassword}
								edge="end"
								data-testid="toggleConfirmPasswordVisibility"
							>
								{renderShowPassword()}
							</IconButton>
						</InputAdornment>
					}
					data-testid="confirmPasswordInput"
				/>

				<StyledTextField
					name="securityQuestion"
					type="text"
					placeholder="Security Question"
					required
					value={securityQuestion}
					onChange={(e) => setSecurityQuestion(e.target.value)}
					data-testid="securityQuestionInput"
				/>

				<StyledTextField
					name="securityAnswer"
					type="text"
					placeholder="Security Answer"
					required
					value={securityAnswer}
					onChange={(e) => setSecurityAnswer(e.target.value)}
					data-testid="securityAnswerInput"
				/>

				<StyledButton
					variant="outlined"
					onClick={handleRegister}
					data-testid="signUpButton"
				>
					Sign Up
				</StyledButton>

			</DialogContent>
		</StyledDialog>
	);
}
