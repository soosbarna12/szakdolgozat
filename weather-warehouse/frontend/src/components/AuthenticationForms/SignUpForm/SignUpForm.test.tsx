import '@testing-library/jest-dom';
import { fireEvent, screen, waitFor } from '@testing-library/react';
import React from 'react';
import { useSignUpQuery } from '../../../hooks/useSignUpQuery';
import { renderWithQueryClient } from '../../../utils/test/renderWithQueryClient';
import { SignUpForm } from './SignUpForm';

jest.mock('../../../hooks/useSignUpQuery', () => ({
    useSignUpQuery: jest.fn(),
}));

describe('AuthenticationForms/SignUpForm', () => {
    const mockOnClose = jest.fn();
    const mockOnRegisterSuccess = jest.fn();
    const mockRefetchRegisterQuery = jest.fn();

    beforeEach(() => {
        jest.clearAllMocks();
        (useSignUpQuery as jest.Mock).mockImplementation(() => ({
            isSuccess: false,
            refetch: mockRefetchRegisterQuery,
        }));
    });

    it('matches SignUpForm component snapshot', () => {
        const component = renderWithQueryClient(
            <SignUpForm open={true} onClose={mockOnClose} onRegisterSuccess={mockOnRegisterSuccess} />
        );

        expect(component).toMatchSnapshot();
    });

    it('resets fields when the dialog is opened', () => {
        const { rerender } = renderWithQueryClient(
            <SignUpForm open={false} onClose={mockOnClose} onRegisterSuccess={mockOnRegisterSuccess} />
        );

        rerender(<SignUpForm open={true} onClose={mockOnClose} onRegisterSuccess={mockOnRegisterSuccess} />);

        expect(screen.getByTestId('usernameInput').querySelector('input')).toHaveValue('');
        expect(screen.getByTestId('passwordInput').querySelector('input')).toHaveValue('');
        expect(screen.getByTestId('confirmPasswordInput').querySelector('input')).toHaveValue('');
        expect(screen.getByTestId('securityQuestionInput').querySelector('input')).toHaveValue('');
        expect(screen.getByTestId('securityAnswerInput').querySelector('input')).toHaveValue('');
    });

    it('handles user input correctly', () => {
        renderWithQueryClient(<SignUpForm open={true} onClose={mockOnClose} onRegisterSuccess={mockOnRegisterSuccess} />);

        const usernameInput = screen.getByTestId('usernameInput').querySelector('input');
        const passwordInput = screen.getByTestId('passwordInput').querySelector('input');
        const confirmPasswordInput = screen.getByTestId('confirmPasswordInput').querySelector('input');
        const securityQuestionInput = screen.getByTestId('securityQuestionInput').querySelector('input');
        const securityAnswerInput = screen.getByTestId('securityAnswerInput').querySelector('input');

        fireEvent.change(usernameInput, { target: { value: 'testuser' } });
        fireEvent.change(passwordInput, { target: { value: 'password123' } });
        fireEvent.change(confirmPasswordInput, { target: { value: 'password123' } });
        fireEvent.change(securityQuestionInput, { target: { value: 'What is your pet’s name?' } });
        fireEvent.change(securityAnswerInput, { target: { value: 'Fluffy' } });

        expect(usernameInput).toHaveValue('testuser');
        expect(passwordInput).toHaveValue('password123');
        expect(confirmPasswordInput).toHaveValue('password123');
        expect(securityQuestionInput).toHaveValue('What is your pet’s name?');
        expect(securityAnswerInput).toHaveValue('Fluffy');
    });

    it('toggles password visibility when the visibility icon is clicked', () => {
        renderWithQueryClient(<SignUpForm open={true} onClose={mockOnClose} onRegisterSuccess={mockOnRegisterSuccess} />);

        const passwordInput = screen.getByTestId('passwordInput').querySelector('input');
        const toggleVisibilityButton = screen.getByTestId('togglePasswordVisibility');

        expect(passwordInput).toHaveAttribute('type', 'password');

        fireEvent.click(toggleVisibilityButton);

        expect(passwordInput).toHaveAttribute('type', 'text');
    });

    it('calls refetchRegisterQuery on sign up button click', async () => {
        renderWithQueryClient(<SignUpForm open={true} onClose={mockOnClose} onRegisterSuccess={mockOnRegisterSuccess} />);

        const signUpButton = screen.getByTestId('signUpButton');

        fireEvent.click(signUpButton);

        await waitFor(() => {
            expect(mockRefetchRegisterQuery).toHaveBeenCalled();
        });
    });

    it('calls onRegisterSuccess and onClose when registration is successful', async () => {
        (useSignUpQuery as jest.Mock).mockImplementation(() => ({
            isSuccess: true,
            refetch: mockRefetchRegisterQuery,
        }));

        renderWithQueryClient(<SignUpForm open={true} onClose={mockOnClose} onRegisterSuccess={mockOnRegisterSuccess} />);

        await waitFor(() => {
            expect(mockOnRegisterSuccess).toHaveBeenCalled();
            expect(mockOnClose).toHaveBeenCalled();
        });
    });
});