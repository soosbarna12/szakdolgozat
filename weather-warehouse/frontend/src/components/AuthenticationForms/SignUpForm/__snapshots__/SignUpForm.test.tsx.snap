// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AuthenticationForms/SignUpForm matches SignUpForm component snapshot 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body
    style="padding-right: 1024px; overflow: hidden;"
  >
    <div
      aria-hidden="true"
    />
    <div
      class="MuiDialog-root MuiModal-root css-17neq5a-MuiModal-root-MuiDialog-root"
      data-testid="signUpDialog"
      role="presentation"
    >
      <div
        aria-hidden="true"
        class="MuiBackdrop-root MuiModal-backdrop css-1cennmq-MuiBackdrop-root-MuiDialog-backdrop"
        style="opacity: 1; webkit-transition: opacity 225ms cubic-bezier(0.4, 0, 0.2, 1) 0ms; transition: opacity 225ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;"
      />
      <div
        data-testid="sentinelStart"
        tabindex="0"
      />
      <div
        class="MuiDialog-container MuiDialog-scrollPaper css-19do60a-MuiDialog-container"
        role="presentation"
        style="opacity: 1; webkit-transition: opacity 225ms cubic-bezier(0.4, 0, 0.2, 1) 0ms; transition: opacity 225ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;"
        tabindex="-1"
      >
        <div
          aria-labelledby="«r0»"
          aria-modal="true"
          class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation24 MuiDialog-paper MuiDialog-paperScrollPaper MuiDialog-paperWidthXs MuiDialog-paperFullWidth css-qopi4n-MuiPaper-root-MuiDialog-paper"
          role="dialog"
          style="--Paper-shadow: 0px 11px 15px -7px rgba(0,0,0,0.2),0px 24px 38px 3px rgba(0,0,0,0.14),0px 9px 46px 8px rgba(0,0,0,0.12);"
        >
          <div
            class="MuiDialogContent-root css-1l3xg6s-MuiDialogContent-root"
          >
            <h5
              class="MuiTypography-root MuiTypography-h5 css-xfkkdu-MuiTypography-root"
              data-testid="signUpTitle"
            >
              Sign Up
            </h5>
            <div
              class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-colorPrimary css-ety0ez-MuiInputBase-root-MuiOutlinedInput-root"
              data-testid="usernameInput"
              required=""
            >
              <input
                class="MuiInputBase-input MuiOutlinedInput-input css-16wblaj-MuiInputBase-input-MuiOutlinedInput-input"
                name="username"
                placeholder="Username"
                required=""
                type="text"
                value=""
              />
              <fieldset
                aria-hidden="true"
                class="MuiOutlinedInput-notchedOutline css-1ll44ll-MuiOutlinedInput-notchedOutline"
              >
                <legend
                  class="css-w4cd9x"
                >
                  <span
                    aria-hidden="true"
                    class="notranslate"
                  >
                    ​
                  </span>
                </legend>
              </fieldset>
            </div>
            <div
              class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-colorPrimary MuiInputBase-adornedEnd css-1ax6i8v-MuiInputBase-root-MuiOutlinedInput-root"
              data-testid="passwordInput"
              required=""
            >
              <input
                class="MuiInputBase-input MuiOutlinedInput-input MuiInputBase-inputAdornedEnd css-1dune0f-MuiInputBase-input-MuiOutlinedInput-input"
                name="password"
                placeholder="Password"
                required=""
                type="password"
                value=""
              />
              <div
                class="MuiInputAdornment-root MuiInputAdornment-positionEnd css-elo8k2-MuiInputAdornment-root"
              >
                <button
                  aria-label="toggle password visibility"
                  class="MuiButtonBase-root MuiIconButton-root MuiIconButton-edgeEnd MuiIconButton-sizeMedium css-1ysp02-MuiButtonBase-root-MuiIconButton-root"
                  data-testid="togglePasswordVisibility"
                  tabindex="0"
                  type="button"
                >
                  <svg
                    aria-hidden="true"
                    class="MuiSvgIcon-root MuiSvgIcon-fontSizeInherit css-1ckov0h-MuiSvgIcon-root"
                    data-testid="VisibilityIcon"
                    focusable="false"
                    viewBox="0 0 24 24"
                  >
                    <path
                      d="M12 4.5C7 4.5 2.73 7.61 1 12c1.73 4.39 6 7.5 11 7.5s9.27-3.11 11-7.5c-1.73-4.39-6-7.5-11-7.5M12 17c-2.76 0-5-2.24-5-5s2.24-5 5-5 5 2.24 5 5-2.24 5-5 5m0-8c-1.66 0-3 1.34-3 3s1.34 3 3 3 3-1.34 3-3-1.34-3-3-3"
                    />
                  </svg>
                </button>
              </div>
              <fieldset
                aria-hidden="true"
                class="MuiOutlinedInput-notchedOutline css-1ll44ll-MuiOutlinedInput-notchedOutline"
              >
                <legend
                  class="css-w4cd9x"
                >
                  <span
                    aria-hidden="true"
                    class="notranslate"
                  >
                    ​
                  </span>
                </legend>
              </fieldset>
            </div>
            <div
              class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-colorPrimary MuiInputBase-adornedEnd css-1ax6i8v-MuiInputBase-root-MuiOutlinedInput-root"
              data-testid="confirmPasswordInput"
              required=""
            >
              <input
                class="MuiInputBase-input MuiOutlinedInput-input MuiInputBase-inputAdornedEnd css-1dune0f-MuiInputBase-input-MuiOutlinedInput-input"
                name="confirmPassword"
                placeholder="Confirm Password"
                required=""
                type="password"
                value=""
              />
              <div
                class="MuiInputAdornment-root MuiInputAdornment-positionEnd css-elo8k2-MuiInputAdornment-root"
              >
                <button
                  aria-label="toggle confirm password visibility"
                  class="MuiButtonBase-root MuiIconButton-root MuiIconButton-edgeEnd MuiIconButton-sizeMedium css-1ysp02-MuiButtonBase-root-MuiIconButton-root"
                  data-testid="toggleConfirmPasswordVisibility"
                  tabindex="0"
                  type="button"
                >
                  <svg
                    aria-hidden="true"
                    class="MuiSvgIcon-root MuiSvgIcon-fontSizeInherit css-1ckov0h-MuiSvgIcon-root"
                    data-testid="VisibilityIcon"
                    focusable="false"
                    viewBox="0 0 24 24"
                  >
                    <path
                      d="M12 4.5C7 4.5 2.73 7.61 1 12c1.73 4.39 6 7.5 11 7.5s9.27-3.11 11-7.5c-1.73-4.39-6-7.5-11-7.5M12 17c-2.76 0-5-2.24-5-5s2.24-5 5-5 5 2.24 5 5-2.24 5-5 5m0-8c-1.66 0-3 1.34-3 3s1.34 3 3 3 3-1.34 3-3-1.34-3-3-3"
                    />
                  </svg>
                </button>
              </div>
              <fieldset
                aria-hidden="true"
                class="MuiOutlinedInput-notchedOutline css-1ll44ll-MuiOutlinedInput-notchedOutline"
              >
                <legend
                  class="css-w4cd9x"
                >
                  <span
                    aria-hidden="true"
                    class="notranslate"
                  >
                    ​
                  </span>
                </legend>
              </fieldset>
            </div>
            <div
              class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-colorPrimary css-ety0ez-MuiInputBase-root-MuiOutlinedInput-root"
              data-testid="securityQuestionInput"
              required=""
            >
              <input
                class="MuiInputBase-input MuiOutlinedInput-input css-16wblaj-MuiInputBase-input-MuiOutlinedInput-input"
                name="securityQuestion"
                placeholder="Security Question"
                required=""
                type="text"
                value=""
              />
              <fieldset
                aria-hidden="true"
                class="MuiOutlinedInput-notchedOutline css-1ll44ll-MuiOutlinedInput-notchedOutline"
              >
                <legend
                  class="css-w4cd9x"
                >
                  <span
                    aria-hidden="true"
                    class="notranslate"
                  >
                    ​
                  </span>
                </legend>
              </fieldset>
            </div>
            <div
              class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-colorPrimary css-ety0ez-MuiInputBase-root-MuiOutlinedInput-root"
              data-testid="securityAnswerInput"
              required=""
            >
              <input
                class="MuiInputBase-input MuiOutlinedInput-input css-16wblaj-MuiInputBase-input-MuiOutlinedInput-input"
                name="securityAnswer"
                placeholder="Security Answer"
                required=""
                type="text"
                value=""
              />
              <fieldset
                aria-hidden="true"
                class="MuiOutlinedInput-notchedOutline css-1ll44ll-MuiOutlinedInput-notchedOutline"
              >
                <legend
                  class="css-w4cd9x"
                >
                  <span
                    aria-hidden="true"
                    class="notranslate"
                  >
                    ​
                  </span>
                </legend>
              </fieldset>
            </div>
            <button
              class="MuiButtonBase-root MuiButton-root MuiButton-outlined MuiButton-outlinedPrimary MuiButton-sizeMedium MuiButton-outlinedSizeMedium MuiButton-colorPrimary MuiButton-root MuiButton-outlined MuiButton-outlinedPrimary MuiButton-sizeMedium MuiButton-outlinedSizeMedium MuiButton-colorPrimary css-vo00dw-MuiButtonBase-root-MuiButton-root"
              data-testid="signUpButton"
              tabindex="0"
              type="button"
            >
              Sign Up
            </button>
          </div>
        </div>
      </div>
      <div
        data-testid="sentinelEnd"
        tabindex="0"
      />
    </div>
  </body>,
  "container": <div
    aria-hidden="true"
  />,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
